packages/lab/src/stepper-input/internal/useSpinner.ts

```diff
@@ -1,33 +1,45 @@
-import { useEffect, useState } from "react";
-
+import { useEffect, useState, SyntheticEvent, useCallback } from "react";
 import { useInterval } from "./useInterval";
 
 const INTERVAL_DELAY = 300;
 
-function useSpinner(activationFn: () => void, isAtLimit: boolean) {
+function useSpinner(
+  activationFn: (event?: SyntheticEvent) => void,
+  isAtLimit: boolean
+) {
   const [buttonDown, setButtonDown] = useState(false);
+  const [event, setEvent] = useState<SyntheticEvent | null>(null);
 
-  const cancelInterval = () => setButtonDown(false);
+  const cancelInterval = () => {
+    setEvent(null);
+  };
 
   useEffect(() => {
-    if (isAtLimit) setButtonDown(false);
+    if (isAtLimit) cancelInterval();
   }, [isAtLimit]);
 
   useEffect(() => {
-    window.addEventListener("keyup", cancelInterval);
     window.addEventListener("mouseup", cancelInterval);
     return () => {
-      window.removeEventListener("keyup", cancelInterval);
       window.removeEventListener("mouseup", cancelInterval);
     };
-  }, []);
-
-  const activate = () => {
-    activationFn();
-    setButtonDown(true);
+  }, [cancelInterval]);
+
+  const activate = (event: SyntheticEvent) => {
+    activationFn(event);
+    if (event.type === "mousedown") {
+      setButtonDown(true);
+      setEvent(event);
+    }
   };
-
-  useInterval(activationFn, buttonDown ? INTERVAL_DELAY : null);
+  useInterval(
+    () => {
+      if (buttonDown && event) {
+        activationFn(event);
```

Comment:
Have you tested whether storing `event` actually worked? e.g. whether accessing event target gives you something? What about event bubbling implications? Can you do some research?